orchagent/dash/dashorch.cpp

```diff
@@ -0,0 +1,299 @@
+#include <cassert>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+#include <exception>
+#include <inttypes.h>
+#include <algorithm>
+
+#include "converter.h"
+#include "dashorch.h"
+#include "macaddress.h"
+#include "orch.h"
+#include "sai.h"
+#include "saiextensions.h"
+#include "swssnet.h"
+#include "tokenize.h"
+
+using namespace std;
+using namespace swss;
+
+extern sai_dash_vip_api_t* sai_dash_vip_api;
+extern sai_dash_direction_lookup_api_t* sai_dash_direction_lookup_api;
+extern sai_object_id_t gSwitchId;
+extern size_t gMaxBulkSize;
+
+DashOrch::DashOrch(DBConnector *db, vector<string> &tableName) : Orch(db, tableName)
+{
+    SWSS_LOG_ENTER();
+}
+
+bool DashOrch::addApplianceEntry(const string& appliance_id, const ApplianceEntry &entry)
+{
+    SWSS_LOG_ENTER();
+
+
+    if (appliance_entries_.find(appliance_id) != appliance_entries_.end())
+    {
+        SWSS_LOG_INFO("Appliance Entry already exists for %s", appliance_id.c_str());
+        return true;
+    }
+
+    sai_vip_entry_t vip_entry;
+    vip_entry.switch_id = gSwitchId;
+    swss::copy(vip_entry.vip, entry.sip);
+    sai_attribute_t appliance_attr;
+    vector<sai_attribute_t> appliance_attrs;
+    sai_status_t status;
+    appliance_attr.id = SAI_VIP_ENTRY_ATTR_ACTION;
+    appliance_attr.value.u32 = SAI_VIP_ENTRY_ACTION_ACCEPT;
+    status = sai_dash_vip_api->create_vip_entry(&vip_entry, 1, &appliance_attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to create vip entry for %s", appliance_id.c_str());
+        task_process_status handle_status = handleSaiCreateStatus((sai_api_t) SAI_API_DASH_VIP, status);
+        if (handle_status != task_success)
+        {
+            return parseHandleSaiStatusFailure(handle_status);
+        }
+    }
+
+    sai_direction_lookup_entry_t direction_lookup_entry;
+    direction_lookup_entry.switch_id = gSwitchId;
+    direction_lookup_entry.vni = entry.vm_vni;
+    appliance_attr.id = SAI_DIRECTION_LOOKUP_ENTRY_ATTR_ACTION;
+    appliance_attr.value.u32 = SAI_DIRECTION_LOOKUP_ENTRY_ACTION_SET_OUTBOUND_DIRECTION;
+    status = sai_dash_direction_lookup_api->create_direction_lookup_entry(&direction_lookup_entry, 1, &appliance_attr);
```

Comment:
magic constant

Reply:
I followed the convention in routeorch.cpp and a few other orchs where the constant "1" is used when there is only a single attribute to be added. A "vector of sai_attribute_t" is not used in such cases to get the size from and the constant "1" is passed as the attribute count. 
I will create a variable attr_count and assign that to 1 instead for better clarity.